receiver/pluginreceiver/plugin.go

```diff
@@ -91,6 +100,29 @@ func (p *Plugin) ApplyDefaults(values map[string]any) map[string]any {
 	return result
 }
 
+func checkExtensions(extensions map[string]any, pluginID string) error {
+	for i, ext := range extensions {
+		c, _ := config.NewComponentIDFromString(i)
+		if c.Type() == "file_storage" {
+			var cfg filestorage.Config
+			err := mapstructure.Decode(ext, &cfg)
+			if err != nil {
+				return err
+			}
+
+			name := strings.ReplaceAll(pluginID, "/", "_")
+			cfg.Directory = filepath.Join(cfg.Directory, name)
+			extensions[i] = map[string]any{"directory": cfg.Directory}
+			err = os.MkdirAll(cfg.Directory, 0750)
```

Comment:
I tried running the tests locally, noticed I ended up with a literal `$OIQ_OTEL_COLLECTOR_HOME` directory in the test dir (it think the Validate tests run this for the default file_logs plugin). It looks like it's because we don't do environment variable substitution, so it just makes the directory literally.

We could do something like:
```go
err = os.MkdirAll(return os.Expand(s, func(s string) string {
    // Allow escaping with $$
    if s == "$" {
        return "$"
    }
    return os.Getenv(str)
}), 0750)
```
This is what the collector does to expand environment variables. I think this will work fine; If the user specifies a directory the parameter will end up expanding twice, unfortunately, but I think that's generally true of all plugin parameters.